src/DataFrame/Operations/Transformations.hs

```diff
@@ -164,3 +164,14 @@ impute columnName value df = case getColumn columnName df of
     Left exception -> throw exception
     Right res      -> res
   _ -> error "Cannot impute to a non-Empty column"
+
+imputeExpr :: forall a. (Columnable a) => T.Text -> Expr a -> DataFrame -> DataFrame
+--imputeExpr = undefined
+imputeExpr columnName expr df =
+  case getColumn columnName df of
+    Nothing -> throw $ ColumnNotFoundException columnName "impute" (map fst $ M.toList $ columnIndices df)
+    Just (OptionalColumn _) ->
+      case expr of
+        Col a -> impute columnName a df
```

Comment:
`Col a` is just a typed reference to a column. It sort of looks like: `data Expr = Col String | ...`. As written, `a === columnName`. You probably want to interpret the expression, get the resulting column then zip it with the result of getColumn.